code/modules/telesci/gps.dm

```diff
@@ -54,6 +54,7 @@ var/list/GPS_list = list()
 	..()
 	if(href_list["tag"] )
 		var/a = input("Please enter desired tag.", name, gpstag) as text
+		if (src.loc != usr || usr.stat) return
```

Comment:
You can only start setting the tag by using the GPS from your hand, if you've dropped it to the ground I'd think it's fair to say the action got interrupted, especially since you're not writing a novel there. I just tried to keep the check consistent with how the action is called.